File: 104. Maximum Depth of Binary Tree/recursion.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    res = 0
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        self.countDepth(root, 1)
        return self.res
        
    def countDepth(self, root, depth):
        if root == None:
            return
        self.res = max(self.res, depth)
        if root.left:
            self.countDepth(root.left, depth+1)
        
        if root.right:
            self.countDepth(root.right, depth+1)

File: 104. Maximum Depth of Binary Tree/recursion.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int res = 0;
    void getDepth(TreeNode* root, int depth) {
        if (root == NULL) return;
        res = max(res, depth);
        if (root->left) {
            getDepth(root->left, depth+1);
        }

        if (root->right) {
            getDepth(root->right, depth+1);
        }
        return;
    }
    int maxDepth(TreeNode* root) {
        getDepth(root, 1);
        return res;
    }
};